apache/spark
# [SPARK-26929][SQL]fix table owner use user instead of principal when create table through spark-sql or beeline

…create table through spark-sql or beeline

## What changes were proposed in this pull request?

fix table owner use user instead of principal when create table through spark-sql
`private val userName = conf.getUser`  will get ugi's userName which is principal info, and i copy the source code into HiveClientImpl, and use `ugi.getShortUserName()` instead of ` ugi.getUserName()`. The owner display correctly.

## How was this patch tested?

1. create a table  in kerberos cluster
2. use "desc formatted tbName" check owner

Please review http://spark.apache.org/contributing.html before opening a pull request.
